plonky2/src/batch_fri/recursive_verifier.rs

```diff
@@ -165,13 +168,24 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
         let mut alpha = ReducingFactorTarget::new(alpha);
         let mut sum = self.zero_extension();
 
-        for (batch, reduced_openings) in instance[index]
+        for (idx, (batch, reduced_openings)) in instance[index]
             .batches
             .iter()
             .zip(&precomputed_reduced_evals.reduced_openings_at_point)
+            .enumerate()
         {
+            // If we are in the zk case, the `R` polynomial (the last polynomials in the first batch) is added to
+            // the batch polynomial independently, without being quotiented. So the final polynomial becomes:
+            // `final_poly = sum_i alpha^(k_i) (F_i(X) - F_i(z_i))/(X-z_i) + alpha^n R(X)`, where `n` is the degree
```

Comment:
I had seen that the indices in the paper started at 1 and assumed that the alpha for the quotient poly was shifted because of R (as in R was the first element in the sum: `R + alpha q_1 + ...`). But upon looking at the code, you are right: we already start with `alpha` and not 1, so I'll remove the multiplication by `alpha^n`.